arrow-select/src/concat.rs

```diff
@@ -1267,4 +1339,123 @@ mod tests {
             "There are duplicates in the value list (the value list here is sorted which is only for the assertion)"
         );
     }
+
+    // Test the simple case of concatenating two RunArrays
+    #[test]
+    fn test_concat_run_array() {
+        // Create simple run arrays
+        let run_ends1 = Int32Array::from(vec![2, 4]);
+        let values1 = Int32Array::from(vec![10, 20]);
+        let array1 = RunArray::try_new(&run_ends1, &values1).unwrap();
+
+        let run_ends2 = Int32Array::from(vec![2, 4]);
```

Comment:
Can you please change the run ends for the second array so they aren't the same as the first? Also it would be good to include a run with length 1

Something like this perhaps:

```suggestion
        let run_ends2 = Int32Array::from(vec![1, 4]);
```